ytfzf

```diff
@@ -720,13 +739,23 @@ get_search_query () {
 			printf "$search_prompt"
 			read -r search_query
 		fi
-		[ -z "$search_query" ] && exit 0
+		if [ -n "$(printf "%s" "$search_query" | grep -Po '!(?:-?[0-9]+|!)')" ]; then
```

Comment:
look at the `-q` option of grep. It will do the operation quietly and return only exit status on match/non-match. So you wouldn't need to check if non empty and directly do something like
```sh
#!/bin/sh
if echo "hi" | grep -q "hi" ; then
	echo "do stuff.."
fi

```